src/shared/utils/hooks/useOnceEffect.ts

```diff
@@ -1,11 +1,19 @@
-import { useEffect } from 'react';
-import type { EffectCallback } from 'react';
+import { EffectCallback, useLayoutEffect, useRef } from 'react';
 
 /**
- * A convenience hook that runs the effect only once, when the component mounts.
+ * A convenience hook that runs the effect only once, when the component mounts, which uses both
+ * a ref and `useLayoutEffect` to ensure that the effect runs only once after the DOM has rendered.
+ *
  * @param effect Some code that you'd normally put in a `useEffect` hook.
  */
 export const useOnceEffect = (effect: EffectCallback) => {
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  useEffect(effect, []);
+  const hasExecutedRef = useRef(false);
+
+  useLayoutEffect(() => {
```

Comment:
These aren't the same thing. They should be separate functions so that the consumer can use them instead of `useEffect` and `useLayoutEffect` as appropriate. Right now we only have one use case for `useOnceLayoutEffect`

Reply:
Oh weird, when I looked at your branch earlier I thought just changed the existing hook rather than introduced a new one. My bad, I must have been misreading it! I was a bit concerned about making a sweeping change as well but thought it was safe/reasonable, deferring to what I thought you'd pushed. I'll create a new hook then!

Reply:
Updated!